.devcontainer/Dockerfile

```diff
@@ -26,10 +26,9 @@ COPY .zshrc /root/.zshrc
 #use zsh as main shell
 RUN chsh -s /usr/bin/zsh root
 
-# install scala -> uncomment one of the following lines
-
-# x86_64
-# RUN curl -fL https://github.com/coursier/coursier/releases/latest/download/cs-x86_64-pc-linux.gz | gzip -d > cs && chmod +x cs && ./cs setup -y && rm -f cs
-
-# For ARM64
-# RUN curl -fL https://github.com/VirtusLab/coursier-m1/releases/latest/download/cs-aarch64-pc-linux.gz | gzip -d > cs && chmod +x cs && ./cs setup -y && rm -f cs
\ No newline at end of file
+ARG TARGETARCH
+RUN if [ "$TARGETARCH" = "amd64" ]; then \
+        curl -fL https://github.com/coursier/coursier/releases/latest/download/cs-x86_64-pc-linux.gz | gzip -d > cs && chmod +x cs && ./cs setup -y && rm -f cs; \
+    else \
+        curl -fL https://github.com/VirtusLab/coursier-m1/releases/latest/download/cs-aarch64-pc-linux.gz | gzip -d > cs && chmod +x cs && ./cs setup -y && rm -f cs; \
+    fi
```

Comment:
I think it will be cleaner to assign the URL within a IF ELSE and then execute `RUN curl $URL`.